src/mon/Monitor.cc

```diff
@@ -5606,6 +5606,13 @@ int Monitor::scrub()
   wait_for_paxos_write();
   scrub_version = paxos->get_version();
 
+  // Create a local copy of scrub_state at this point because
+  // it might be possible the scrub_state is reset via bootstrap
+  // code path and daemon is crashed.
+  std::shard_ptr<ScrubState> local_scrub_state = scrub_state;
+  if (nullptr == local_scrub_state) {
```

Comment:
`scrub_version` and `scrub_results` are also modified by `Monitor::scrub_reset()`, so similar race affects them.

Perhaps we could group these members to e.g. `ScrubContext`  and manage a pointer with the atomic shared ptr. Alternatively, a lock can be used but – as the scrub can take some time – better to have a dedicated one (like `Monitor::auth_lock`).

Reply:
Ack